spec/lib/artifact.spec.js

```diff
@@ -0,0 +1,292 @@
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.upload(content).then(() => {
+                expect(bluebird.fromCallback).toHaveBeenCalled();
+            });
+        });
+        it("calls requestOptions.createOptions()", () => {
+            requestMock.post.andCallFake((options, resolver) => {
+                return {
+                    form: () => {
+                        return {
+                            append: () => {
+                                return resolver(null, {
+                                    headers: {
+                                        location: "someLocation"
+                                    }
+                                });
+                            }
+                        };
+                    }
+                };
+            });
+
+            return instance.upload(content).then(() => {
+                expect(requestOptionsMock.createOptions).toHaveBeenCalled();
+            });
+        });
+        it("calls request.post()", () => {
+            requestMock.post.andCallFake((options, resolver) => {
+                return {
+                    form: () => {
+                        return {
+                            append: () => {
+                                return resolver(null, {
+                                    headers: {
+                                        location: "someLocation"
+                                    }
+                                });
+                            }
+                        };
+                    }
+                };
+            });
+
+            return instance.upload(content).then(() => {
+                expect(requestMock.post).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            responseHandlerMock.isErrorCode.andReturn(true);
+            spyOn(bluebird, "fromCallback").andReturn(bluebird.resolve({
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.upload(content).then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.resolveLink", () => {
+            spyOn(bluebird, "fromCallback").andReturn(bluebird.resolve({
+                headers: {
+                    location: "someLocation"
+                }
+            }));
+
+            return instance.upload(content).then(() => {
+                expect(responseHandlerMock.resolveLink).toHaveBeenCalled();
+            });
+        });
+    });
+    describe(".uploadFromFile()", () => {
+        beforeEach(() => {
+            responseHandlerMock.isErrorCode.andReturn(true);
+        });
+        it("calls bluebird.fromCallback()", () => {
```

Comment:
From what I understand, this was the easiest way to handle `request.post`'s callback param. `fromCallback` provides a `resolver` which can be passed into a function in place of a callback.